### PR TITLE
fix input device name detection for usb devices (bsc #989633)

### PR DESCRIPTION
**Pull request by wfeldt**
The link to input devices has been moved one level deeper into the sysfs tree
for usb devices.

### CODE REVIEW

**Comment by wfeldt** [comment]
A big part of the patch is just indentation changes.

File: src/hd/usb.c [hunk 1]
```diff
@@ -569,41 +569,48 @@ void add_input_dev(hd_data_t *hd_data, char *name)
...
-        t = NULL;
-        str_printf(&t, 0, "/dev/input/%s", sf_cdev_name);
+    /* find device (matching sysfs path) */
+    hd = hd_find_sysfs_id(hd_data, s);
+
+    /* not found? - retry one level up */
+    if(!hd) {
+      char *ns = new_str(s), *nt;
+      if((nt = strrchr(ns, '/'))) {
+        *nt = 0;
```
Comment (jreidinger) [inline src/hd/usb.c]: why it is needed? it is local variable and used only once here.

Comment (wfeldt) [inline src/hd/usb.c]: Sorry, what do yo mean?

Comment (jreidinger) [inline src/hd/usb.c]: ok, now I get code. quite obfuscated.
what about something like
```
nt = strrchr(s, '/');
unsigned substr_len = nt - s;
char ns = malloc(substr_len + 1);
memcpy(ns, s, substr_len);
ns[substr_len] = 0;
```

this way it is a bit longer, but for me more readable. as it is obvious that it copy substring to new memory location. Also can mem_free dealocate whole string even when earlier finished? ( I do not check implementation if it rely on string or if it use memory allocation map )

Comment (wfeldt) [inline src/hd/usb.c]: I can't follow you; your implementation is way longer, not taking into account that there might be no '/' in the string, and much more obscure to me.

In my implementation it just duplicates a string and cuts off the end - perfectly clear.

`mem_free` works just like `free` - it doesn't care what's in the memory blob. Josef, you should do more C...

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**